### PR TITLE
[DO-983] Export fop

### PR DESCRIPTION
**Pull request by Iva-khar**
By api like: `/api/fop/xlsx/?registration_date_after=2021-03-10&registration_date_before=2021-04-05&status=2` uploads .xlsx file to s3 like:
Screenshot from 2021-05-08 13-35-19

### CODE REVIEW

File: data_ocean/export.py [hunk 1]
```diff
@@ -41,13 +41,14 @@ def export(params, export_dict, model_path, filterset_module, user_id):
...
             for record in queryset:
-
                 row_num += 1
                 cell = worksheet.cell(row=row_num, column=col_num)
                 cell.alignment = Alignment(vertical='top', wrap_text=True)
                 cell_value = getattr(record, column_properties[0])
                 if isinstance(cell_value, datetime):
                     cell_value = cell_value.replace(tzinfo=None)
+                elif not isinstance(cell_value, (str, int)):
+                    cell_value = str(cell_value)
```
Comment (roman-tiukh) [inline data_ocean/export.py:51]: Nope, its bad. You have other types what can be supported with openpyxl, for example `timedelta`.

File: business_register/views/fop_views.py [hunk 2]
```diff
@@ -25,3 +31,28 @@ class FopViewSet(RegisterViewMixin,
...
+            'Registration Date': ['registration_date', 18],
+            'Termination Date': ['termination_date', 18],
+            'Created Date': ['created_at', 19],
+            'Updated Date': ['updated_at', 19],
+            'Authority': ['authority', 36]
+        }
+        from data_ocean.tasks import export_to_s3
+        export_to_s3.delay(request.GET, export_dict, 'business_register.Fop',
+                           'business_register.filters.FopExportFilterSet', request.user.id)
+        from django.utils.translation import gettext_lazy as _
```
Comment (roman-tiukh) [inline business_register/views/fop_views.py:54]: move this import to top

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
Duplicated code
image

**Comment by Iva-khar** [comment]
> Duplicated code
> image

Different fields. Messages chenged (also different)

**Comment by Iva-khar** [comment]
> Duplicated code
> image

Maked abstract form

File: data_ocean/export.py (hunk 1 above)
Comment (Iva-khar) [reply data_ocean/export.py]: Rewrited.

File: business_register/views/fop_views.py (hunk 2 above)
Comment (Iva-khar) [reply business_register/views/fop_views.py]: done

**Review by roman-tiukh — APPROVED** [review]

File: data_ocean/export.py [hunk 3]
```diff
@@ -41,13 +39,14 @@ def export(params, export_dict, model_path, filterset_module, user_id):
...
             cell.fill = PatternFill(bgColor='0033CCCC', fill_type="solid")
             for record in queryset:
-
                 row_num += 1
                 cell = worksheet.cell(row=row_num, column=col_num)
                 cell.alignment = Alignment(vertical='top', wrap_text=True)
                 cell_value = getattr(record, column_properties[0])
                 if isinstance(cell_value, datetime):
                     cell_value = cell_value.replace(tzinfo=None)
+                elif isinstance(cell_value, (Authority, Status)):
```
[the remark this answers is not in the document]
Comment (Iva-khar) [reply data_ocean/export.py:48]: of course

[the remark this answers is not in the document]
Comment (Iva-khar) [reply data_ocean/export.py]: done

**Pull request merged**